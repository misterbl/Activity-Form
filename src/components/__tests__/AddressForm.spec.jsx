import * as React from "react";
import { shallow } from "enzyme";
import AddressForm from "../AddressForm";
import formikRenderMock from "../../testMocks/formikRender.mock";

describe("AddressForm", () => {
  const props = {
    ...formikRenderMock,
    values: {
      buildingUnit: "buildingUnit",
      buildingName: "buildingName",
      streetNumber: "streetNumber",
      streetName: "streetName",
      town: "town",
      postcode: "postcode"
    },
    initialValues: { teamName: "" },
    isInFlight: false,
    isError: false
  };
  const wrapper = shallow(<AddressForm {...props} />);
  it("matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
