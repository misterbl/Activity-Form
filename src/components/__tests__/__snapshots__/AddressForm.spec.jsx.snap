// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressForm matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddressForm
    dirty={false}
    errors={Object {}}
    handleBlur={[Function]}
    handleChange={[Function]}
    handleReset={[Function]}
    handleSubmit={[Function]}
    isSubmitting={false}
    isValid={false}
    isValidating={false}
    registerField={[Function]}
    resetForm={[Function]}
    setError={[Function]}
    setErrors={[Function]}
    setFieldError={[Function]}
    setFieldTouched={[Function]}
    setFieldValue={[Function]}
    setFormikState={[Function]}
    setStatus={[Function]}
    setSubmitting={[Function]}
    setTouched={[Function]}
    setValues={[Function]}
    submitCount={0}
    submitForm={[Function]}
    touched={Object {}}
    unregisterField={[Function]}
    validateField={[Function]}
    validateForm={[Function]}
    values={
      Object {
        "buildingName": "buildingName",
        "buildingUnit": "buildingUnit",
        "postcode": "postcode",
        "streetName": "streetName",
        "streetNumber": "streetNumber",
        "town": "town",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          className="font-weight-bold d-flex justify-content-between"
          htmlFor="buildingUnit"
        >
          <span>
            Building Unit
          </span>
          <span
            className="font-weight-lighter text-black-50"
          >
            Optional
          </span>
        </label>,
        <input
          className="form-control mb-4"
          id="buildingUnit"
          name="buildingUnit"
          onChange={[Function]}
          type="text"
          value="buildingUnit"
        />,
        <label
          className="font-weight-bold m-0 d-flex justify-content-between"
          htmlFor="buildingName"
        >
          <span>
            Building Name
          </span>
          <span
            className="font-weight-lighter text-black-50"
          >
            Optional
          </span>
        </label>,
        <input
          className="form-control mb-4"
          id="buildingName"
          name="buildingName"
          onChange={[Function]}
          type="text"
          value="buildingName"
        />,
        <label
          className="font-weight-bold"
          htmlFor="streetNumber"
        >
          Street Number
        </label>,
        <input
          aria-invalid={false}
          className="form-control mb-4"
          id="streetNumber"
          name="streetNumber"
          onChange={[Function]}
          type="text"
          value="streetNumber"
        />,
        false,
        <label
          className="font-weight-bold"
          htmlFor="streetName"
        >
          Street Name
        </label>,
        <input
          aria-invalid={false}
          className="form-control mb-4"
          id="streetName"
          name="streetName"
          onChange={[Function]}
          type="text"
          value="streetName"
        />,
        false,
        <label
          className="font-weight-bold"
          htmlFor="town"
        >
          Town
        </label>,
        <input
          aria-invalid={false}
          className="form-control mb-4"
          id="town"
          name="town"
          onChange={[Function]}
          type="text"
          value="town"
        />,
        false,
        <label
          className="font-weight-bold"
          htmlFor="postcode"
        >
          Postcode
        </label>,
        <input
          aria-invalid={false}
          className="form-control mb-4"
          id="postcode"
          name="postcode"
          onChange={[Function]}
          type="text"
          value="postcode"
        />,
        false,
        <div
          className="button-container bg-white d-flex flex-row-reverse py-2"
        >
          <button
            className="btn next-button ml-3"
            disabled={false}
            id="next-button"
            type="submit"
          >
            Next
          </button>
          <button
            className="btn back-button"
            id="back-button"
            type="button"
          >
            Back
          </button>
        </div>,
      ],
      "className": "form-group mb-0",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              Building Unit
            </span>,
            <span
              className="font-weight-lighter text-black-50"
            >
              Optional
            </span>,
          ],
          "className": "font-weight-bold d-flex justify-content-between",
          "htmlFor": "buildingUnit",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Building Unit",
            },
            "ref": null,
            "rendered": "Building Unit",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Optional",
              "className": "font-weight-lighter text-black-50",
            },
            "ref": null,
            "rendered": "Optional",
            "type": "span",
          },
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "form-control mb-4",
          "id": "buildingUnit",
          "name": "buildingUnit",
          "onChange": [Function],
          "type": "text",
          "value": "buildingUnit",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              Building Name
            </span>,
            <span
              className="font-weight-lighter text-black-50"
            >
              Optional
            </span>,
          ],
          "className": "font-weight-bold m-0 d-flex justify-content-between",
          "htmlFor": "buildingName",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Building Name",
            },
            "ref": null,
            "rendered": "Building Name",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Optional",
              "className": "font-weight-lighter text-black-50",
            },
            "ref": null,
            "rendered": "Optional",
            "type": "span",
          },
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "form-control mb-4",
          "id": "buildingName",
          "name": "buildingName",
          "onChange": [Function],
          "type": "text",
          "value": "buildingName",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Street Number",
          "className": "font-weight-bold",
          "htmlFor": "streetNumber",
        },
        "ref": null,
        "rendered": "Street Number",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-invalid": false,
          "className": "form-control mb-4",
          "id": "streetNumber",
          "name": "streetNumber",
          "onChange": [Function],
          "type": "text",
          "value": "streetNumber",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Street Name",
          "className": "font-weight-bold",
          "htmlFor": "streetName",
        },
        "ref": null,
        "rendered": "Street Name",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-invalid": false,
          "className": "form-control mb-4",
          "id": "streetName",
          "name": "streetName",
          "onChange": [Function],
          "type": "text",
          "value": "streetName",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Town",
          "className": "font-weight-bold",
          "htmlFor": "town",
        },
        "ref": null,
        "rendered": "Town",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-invalid": false,
          "className": "form-control mb-4",
          "id": "town",
          "name": "town",
          "onChange": [Function],
          "type": "text",
          "value": "town",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Postcode",
          "className": "font-weight-bold",
          "htmlFor": "postcode",
        },
        "ref": null,
        "rendered": "Postcode",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-invalid": false,
          "className": "form-control mb-4",
          "id": "postcode",
          "name": "postcode",
          "onChange": [Function],
          "type": "text",
          "value": "postcode",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="btn next-button ml-3"
              disabled={false}
              id="next-button"
              type="submit"
            >
              Next
            </button>,
            <button
              className="btn back-button"
              id="back-button"
              type="button"
            >
              Back
            </button>,
          ],
          "className": "button-container bg-white d-flex flex-row-reverse py-2",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Next",
              "className": "btn next-button ml-3",
              "disabled": false,
              "id": "next-button",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Next",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Back",
              "className": "btn back-button",
              "id": "back-button",
              "onClick": undefined,
              "type": "button",
            },
            "ref": null,
            "rendered": "Back",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            className="font-weight-bold d-flex justify-content-between"
            htmlFor="buildingUnit"
          >
            <span>
              Building Unit
            </span>
            <span
              className="font-weight-lighter text-black-50"
            >
              Optional
            </span>
          </label>,
          <input
            className="form-control mb-4"
            id="buildingUnit"
            name="buildingUnit"
            onChange={[Function]}
            type="text"
            value="buildingUnit"
          />,
          <label
            className="font-weight-bold m-0 d-flex justify-content-between"
            htmlFor="buildingName"
          >
            <span>
              Building Name
            </span>
            <span
              className="font-weight-lighter text-black-50"
            >
              Optional
            </span>
          </label>,
          <input
            className="form-control mb-4"
            id="buildingName"
            name="buildingName"
            onChange={[Function]}
            type="text"
            value="buildingName"
          />,
          <label
            className="font-weight-bold"
            htmlFor="streetNumber"
          >
            Street Number
          </label>,
          <input
            aria-invalid={false}
            className="form-control mb-4"
            id="streetNumber"
            name="streetNumber"
            onChange={[Function]}
            type="text"
            value="streetNumber"
          />,
          false,
          <label
            className="font-weight-bold"
            htmlFor="streetName"
          >
            Street Name
          </label>,
          <input
            aria-invalid={false}
            className="form-control mb-4"
            id="streetName"
            name="streetName"
            onChange={[Function]}
            type="text"
            value="streetName"
          />,
          false,
          <label
            className="font-weight-bold"
            htmlFor="town"
          >
            Town
          </label>,
          <input
            aria-invalid={false}
            className="form-control mb-4"
            id="town"
            name="town"
            onChange={[Function]}
            type="text"
            value="town"
          />,
          false,
          <label
            className="font-weight-bold"
            htmlFor="postcode"
          >
            Postcode
          </label>,
          <input
            aria-invalid={false}
            className="form-control mb-4"
            id="postcode"
            name="postcode"
            onChange={[Function]}
            type="text"
            value="postcode"
          />,
          false,
          <div
            className="button-container bg-white d-flex flex-row-reverse py-2"
          >
            <button
              className="btn next-button ml-3"
              disabled={false}
              id="next-button"
              type="submit"
            >
              Next
            </button>
            <button
              className="btn back-button"
              id="back-button"
              type="button"
            >
              Back
            </button>
          </div>,
        ],
        "className": "form-group mb-0",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                Building Unit
              </span>,
              <span
                className="font-weight-lighter text-black-50"
              >
                Optional
              </span>,
            ],
            "className": "font-weight-bold d-flex justify-content-between",
            "htmlFor": "buildingUnit",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Building Unit",
              },
              "ref": null,
              "rendered": "Building Unit",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Optional",
                "className": "font-weight-lighter text-black-50",
              },
              "ref": null,
              "rendered": "Optional",
              "type": "span",
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-control mb-4",
            "id": "buildingUnit",
            "name": "buildingUnit",
            "onChange": [Function],
            "type": "text",
            "value": "buildingUnit",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                Building Name
              </span>,
              <span
                className="font-weight-lighter text-black-50"
              >
                Optional
              </span>,
            ],
            "className": "font-weight-bold m-0 d-flex justify-content-between",
            "htmlFor": "buildingName",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Building Name",
              },
              "ref": null,
              "rendered": "Building Name",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Optional",
                "className": "font-weight-lighter text-black-50",
              },
              "ref": null,
              "rendered": "Optional",
              "type": "span",
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-control mb-4",
            "id": "buildingName",
            "name": "buildingName",
            "onChange": [Function],
            "type": "text",
            "value": "buildingName",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Street Number",
            "className": "font-weight-bold",
            "htmlFor": "streetNumber",
          },
          "ref": null,
          "rendered": "Street Number",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-invalid": false,
            "className": "form-control mb-4",
            "id": "streetNumber",
            "name": "streetNumber",
            "onChange": [Function],
            "type": "text",
            "value": "streetNumber",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Street Name",
            "className": "font-weight-bold",
            "htmlFor": "streetName",
          },
          "ref": null,
          "rendered": "Street Name",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-invalid": false,
            "className": "form-control mb-4",
            "id": "streetName",
            "name": "streetName",
            "onChange": [Function],
            "type": "text",
            "value": "streetName",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Town",
            "className": "font-weight-bold",
            "htmlFor": "town",
          },
          "ref": null,
          "rendered": "Town",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-invalid": false,
            "className": "form-control mb-4",
            "id": "town",
            "name": "town",
            "onChange": [Function],
            "type": "text",
            "value": "town",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Postcode",
            "className": "font-weight-bold",
            "htmlFor": "postcode",
          },
          "ref": null,
          "rendered": "Postcode",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-invalid": false,
            "className": "form-control mb-4",
            "id": "postcode",
            "name": "postcode",
            "onChange": [Function],
            "type": "text",
            "value": "postcode",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="btn next-button ml-3"
                disabled={false}
                id="next-button"
                type="submit"
              >
                Next
              </button>,
              <button
                className="btn back-button"
                id="back-button"
                type="button"
              >
                Back
              </button>,
            ],
            "className": "button-container bg-white d-flex flex-row-reverse py-2",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Next",
                "className": "btn next-button ml-3",
                "disabled": false,
                "id": "next-button",
                "type": "submit",
              },
              "ref": null,
              "rendered": "Next",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Back",
                "className": "btn back-button",
                "id": "back-button",
                "onClick": undefined,
                "type": "button",
              },
              "ref": null,
              "rendered": "Back",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
